### PR TITLE
LibGUI: Show human readable size in filemanager

### LINKED ISSUE #5938: FileManager: Display file size in a human readable format
It would be nice to display the file size in a human readable format.

I took a quick look at it and made `FileSystemModel::data` return a formatted `String` instead of an `int` which would solve the issue. But the range selection size display use the same function to then calculate the total size of the files in the selection, so such a change would break the UI. There is maybe a better way to display the size in the `FileSystemModel`?

Current implementation:

```c++
Variant FileSystemModel::data(const ModelIndex& index, ModelRole role) const
...
    if (role == ModelRole::Display) {
...
        case Column::Size:
            return (int)node.size;
```

image

Proposal (with broken range select):

```c++
Variant FileSystemModel::data(const ModelIndex& index, ModelRole role) const
...
    if (role == ModelRole::Display) {
...
        case Column::Size:
            return human_readable_size(node.size);
```

image

### PR DESCRIPTION
**Pull request by guerinoni**
Th size is visible only in TableView mode.
This fixes #5938.

### CODE REVIEW

**Comment by linusg** [comment]
As mentioned in #5938, this breaks the size of selected files information in the status bar.

**Comment by linusg** [comment]
Here's how to fix it:

```diff
--- a/Userland/Applications/FileManager/DirectoryView.cpp
+++ b/Userland/Applications/FileManager/DirectoryView.cpp
@@ -413,10 +413,8 @@ void DirectoryView::update_statusbar()
     size_t selected_byte_count = 0;
 
     current_view().selection().for_each_index([&](auto& index) {
-        auto& model = *current_view().model();
-        auto size_index = model.index(index.row(), GUI::FileSystemModel::Column::Size, model.parent_index(index));
-        auto file_size = size_index.data().to_i32();
-        selected_byte_count += file_size;
+        auto& node = this->node(index);
+        selected_byte_count += node.size;
     });
 
     StringBuilder builder;
```

File: Userland/Libraries/LibGUI/FileSystemModel.cpp [hunk 1]
```diff
@@ -439,7 +440,7 @@ Variant FileSystemModel::data(const ModelIndex& index, ModelRole role) const
         case Column::Name:
             return node.name;
         case Column::Size:
-            return (int)node.size;
+            return AK::human_readable_size(node.size);
```
Comment (linusg) [inline Userland/Libraries/LibGUI/FileSystemModel.cpp:443]: please drop the AK namespace prefix here, this has a `using` directive:

```suggestion
            return human_readable_size(node.size);
```

Comment (guerinoni) [reply Userland/Libraries/LibGUI/FileSystemModel.cpp]: Thanks for the suggestion! I fixed it!

**Comment by guerinoni** [comment]
@linusg many thanks for the accurate review and suggestions. If this is ok I will autosquash in 1 commit

**Comment by linusg** [comment]
Yes, please squash both fixup commits into the initial one :)

**Review by linusg — APPROVED** [review]

**Pull request merged**